app_dart/test/service/cache_service_test.dart

```diff
@@ -118,6 +119,28 @@ void main() {
       expect(valueAfterPurge, isNull);
     });
 
+    test('set blocks read attempt', () async {
+      const String testKey = 'abc';
+      final Uint8List expectedValue = Uint8List.fromList('123'.codeUnits);
+
+      unawaited(cache.set(testSubcacheName, testKey, expectedValue));
```

Comment:
Instead, doing `final cacheWrite = cache.set(testSubcacheName, testKey, expectedValue);` will prevent race conditions with timers left pending.

```dart
final cacheWrite = cache.set(testSubcacheName, testKey, expectedValue);
// Do the tests

await cacheWrite; // Finish pending futures.
```

Reply:
Ah okay. Cool. Thank you.